app/models/group.rb

```diff
@@ -3,13 +3,16 @@ class Group
   attr_accessor :description
   attr_accessor :uuid
   attr_writer   :users
+  attr_writer   :roles
 
   def self.find(uuid)
     group = nil
     Insights::API::Common::RBAC::Service.call(RBACApiClient::GroupApi) do |api|
       group = from_raw(api.get_group(uuid))
     end
     group
+  rescue RBACApiClient::ApiError => e
+    raise unless e.code == 404
```

Comment:
@hsong-rh Just an FYI

This is how the generated clients set the error code

https://github.com/RedHatInsights/insights-rbac-api-client-ruby/blob/a1def54141dc0ad2376af0035837d1741781c919/lib/rbac-api-client/api_client.rb#L58

the code is really HTTP status code so if the server cannot be reached or there is some other networking issue the code will be 0

Reply:
Only changed to return `nil` when group is not found, instead of raising exception, and still keep all other errors as they are.